### DISCUSSION TITLE
User only Read

### DISCUSSION THREAD

**Discussion by Salomoniki** [immich-app/immich#8236] [Feature Request]
### The feature

Brilliant app.  There are only two things missing to complete happiness.  An account option (user) with read-only access would be useful.  No backup, editing, etc. For family members. The second thing is translation into other languages, but that's a detail

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by Yohan-Janolin** [comment]
I second this idea. It would be great to be able to define some allowed/forbidden actions for each account (or a group of accounts). For example, everything is forbidden except creating albums and download photos (which would be ideal for sharing photos with the family for example)

**Comment by nsu700** [comment]
I am also looking for this feature, especially for a family user, most of the time we will upload image through phone or laptop, and may view it in the TV, at that time, a readonly user for TV is needed

**Comment by max-kibosh** [comment]
this would be amazing, especially with group-level access functionality!
that way we could share albums with an entire group, for example, and the respective access settings take care of the rest, instead of indivually sharing albums with users and setting rights one by one.

**Comment by josx2** [comment]
Another vote for this feature 🤘  would love to be able to have family join albums, but not have ability to upload

**Comment by jgiszczak** [comment]
Read only could possibly be implemented as quota 0 GiB and have -1 GiB mean Unlimited.

But an explicit toggle in the Create User dialog and in user settings is probably better UI.

**Comment by Hossy** [comment]
+1 on this.  I want to be able to add friends and family without allowing them to eat up my server storage.  No uploads, just view the albums I share (and they can then get notified when a new photo is added).

**Comment by jh724186** [comment]
+1 Here. Elderly relative (90+), computer literate but can accidentally delete photos. Yes, I can recover for them, but some way of locking down their account once their photos have been uploaded would be very useful. Either via the user or admin settings.

**Comment by cheesebo** [comment]
Another vote from me. This would be very useful to allow sharing of older photos with the wider family.

**Comment by InventorB** [comment]
Sounds like a great feature addition.

**Comment by Dezc** [comment]
+1 too !

**Comment by papercutter0324** [comment]
To be honest, I'm surprised this isn't already a thing. When I decided to host my own photos and came across Immich, I kinda assumed this was going to be a basic feature. Have users who can only view, so I can share my photos with family on the other side of the world. Have groups to set which albums are visible to which people. Seems like a basic user management concept.

So, another +1 for this feature requestl

**Comment by indam** [comment]
Since the fine permission control has already been supported in [version 1.135.0](https://github.com/immich-app/immich/releases/tag/v1.135.0), it should not be that difficult to create the Read-Only user. This application is fantastic, but lack of this feature is my final stumbling block in installing immich on my parents' phones.

Please support this feature.

**Comment by bo0tzz** [comment]
It's now possible to stop a user from uploading by setting their quota to 0.

**Reply by jh724186** [reply]
> It's now possible to stop a user from uploading by setting their quota to 0.

Does setting the quota to 0 also make the existing files read-only (which is the feature being requested) ?

**Reply by WilcoSnel** [reply]
For me, my family should be able to upload into a specific album only. Setting this to 0 would prevent that. Authorisation model is required for this. Hopefully this can be worked on in the future.

**Comment by cheesebo** [comment]
Unless I'm missing something, this isn't a read only user - it just gives a user access to shared albums only. The user has no access to view photos except via an album, no sharing of people, location or timeline.
This is not a read only user in the sense of the requests here. The requests here are to share the whole photo library with other users who can enjoy all the features of Immich, such as location, people and timeline - but not upload or edit photos.
The solution given is really no better than giving a user a shared link to an album.

**Comment by laurepro** [comment]
Great app. Regarding the quota 0 user to simulate a read-only user, it would be easy to remove the menu entries and access routes to the update and share features for quota 0 users, to avoid loading error notifications for example. The home page should also be /albums and not /photos. It is up to the main user to share albums to make them visible to quota 0 users.

**Comment by tampadesignr** [comment]
this will be inevitable so lets get this made now, lol
a way to make a selection of pictures and build a static website with that information, 
so you can share it on a github pages account, easy as 123
-- all these generators, for content, design elements, etc.... the pieces are all here, literally just need a hero to build it
---- simple request--- select 20 pictures and make a static page that can be hosted anywhere, anytime
